lib/tokenize.cpp

```diff
@@ -10117,7 +10117,7 @@ void Tokenizer::simplifyKeyword()
             if (tok->str() == "restrict") {
                 // check if restrict is used as a type qualifier
                 if (Token::Match(tok->previous(), "*|[ restrict %name%") ||
-                    Token::Match(tok->next(), "* %name%"))
+                    (Token::Match(tok->next(), "* %name%") && tok->strAt(2) != "restrict"))
```

Comment:
that was not a proper fix for it.. the name could be anything I just happened to reuse `restrict`

     void foo(int restrict) { a = buf[restrict * x]; }

Imho this code: `(int restrict)` should mean that we don't treat restrict as a keyword in other places. Are you against a 2-pass simplification? An alternative is to just add a flag so that after a declaration where restrict is not a keyword, this simplification will never be used.. I feel that would be pretty good but well a 2-pass simplification would feel even more robust.

Reply:
This approach isn't going to work in all cases so I'm closing this PR.

A 2-pass solution would be better but I don't have the time to implement it.  The same c89 in c99 problem also exists for `inline`.